src/components/pages/address/address-events.tsx

```diff
@@ -1,13 +1,108 @@
-import { Address } from "viem";
-import { Card, CardHeader, CardContent } from "@/components/ui/card";
-
-const AddressEvents = ({ address }: { address: Address }) => {
-  return (
-    <Card>
-      <CardHeader>Latest 25 Contract Events</CardHeader>
-      <CardContent>CONTRACT EVENTS</CardContent>
-    </Card>
-  );
+
+import Link from "next/link";
+import { Address, Block } from "viem";
+import { abiFromBytecode, whatsabi } from "@shazow/whatsabi";
+
+import { l2PublicClient } from "@/lib/chains";
+import { formatEventLog, formatTimestamp } from "@/lib/utils";
+
+import { Table, TableHeader, TableRow, TableHead, TableBody, TableCell } from "@/components/ui/table";
+import { DecodedLogDisplay } from "@/components/ui/decodedLog-display"
+
+import { ABIEventExtended, FormattedLog } from "@/interfaces";
+
+const AddressEvents = async ({ address }: { address: Address }) => {
+  try {
+    const latestBlockNumber: bigint = await l2PublicClient.getBlockNumber();
+    const fromBlock: bigint = latestBlockNumber - BigInt(10);
+    const [bytecode, logs] = await Promise.all([
+      l2PublicClient.getCode({ address }) as Promise<string>,
+      l2PublicClient.getLogs({ address, fromBlock, toBlock: 'latest' }),
+    ]);
+
+    const signatureLookup = new whatsabi.loaders.OpenChainSignatureLookup();
+    const abi = abiFromBytecode(bytecode) as ABIEventExtended[];
+
+    const blockTimestamps = await Promise.all(logs.map(async log => {
+      const block: Block = await l2PublicClient.getBlock({ blockNumber: log.blockNumber });
+      return block.timestamp;
+    }));
+
+    const decodedLogs: FormattedLog[] = await Promise.all(logs.map(async (log, index) => {
+      const formattedLog = await formatEventLog(log, abi, signatureLookup);
+      return {
+        ...formattedLog,
+        transactionHash: log.transactionHash,
+        blockNumber: log.blockNumber,
+        timestamp: {
```

Comment:
I'm not sure you should use formatTimestamp here, I think it's better to just get the timestamp from the block and call formatTimestamp in the render function.